templates/pkg/resource/references.go.tpl

```diff
@@ -18,10 +18,28 @@ import (
 	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
 {{ end -}}
 	acktypes "github.com/aws-controllers-k8s/runtime/pkg/types"
-
 	svcapitypes "github.com/aws-controllers-k8s/{{ .ServicePackageName }}-controller/apis/{{ .APIVersion }}"
+{{ $servicePackageName := .ServicePackageName -}}
+{{ $apiVersion := .APIVersion -}}
+{{ if .CRD.HasReferenceFields -}}
+{{ range $fieldName, $field := .CRD.Fields -}}
+{{ if and $field.HasReference (not (eq $field.ReferencedServiceName $servicePackageName)) -}}
+    {{ $field.ReferencedServiceName }}apitypes "github.com/aws-controllers-k8s/{{ $field.ReferencedServiceName }}-controller/apis/{{ $apiVersion }}"
```

Comment:
I presume for now we are requiring the controller's `go.mod` files to be manually updated with the imported referenced service controller's package?

Reply:
yes. I also mentioned it in the proposal PR. https://github.com/aws-controllers-k8s/community/pull/1092/files